OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/CourseModalDialogFragment.kt

```diff
@@ -121,6 +133,31 @@ class CourseModalDialogFragment : DialogFragment() {
             })
     }
 
+    private fun fetchProductPrice() {
+        ProductManager.getProductByCourseId(courseId)?.let {
+            billingProcessor?.querySyncDetails(
+                productId = it
+            ) { _, skuDetails ->
+                val skuDetail = skuDetails?.get(0)
+                if (skuDetail?.sku == it) {
+                    binding.layoutUpgradeBtn.btnUpgrade.text =
+                        ResourceUtil.getFormattedString(
+                            resources,
+                            R.string.label_upgrade_course_button,
+                            AppConstants.PRICE,
+                            skuDetail.price
+                        ).toString()
+                }
+                // The app get the sku details instantly, so add some wait to perform
+                // animation at least one cycle.
+                binding.shimmerViewContainer.postDelayed({
```

Comment:
IMO, shimmer shouldn't stop until we get the course price. 🤔
So in that case it should be part of if condition.  

We need to handle else case in this scenario to i.e show error dialog maybe